platform/security/common/src/main/java/ddf/security/samlp/impl/EntityInformation.java

```diff
@@ -102,14 +102,19 @@ public ServiceInfo getAssertionConsumerService(
       AuthnRequest request, Binding preferred, Integer index) {
     ServiceInfo si = null;
     if (request != null && request.getAssertionConsumerServiceURL() != null) {
-      return getAssertionConsumerServiceInfoByUrl(request);
+      si = getAssertionConsumerServiceInfoByUrl(request);
+      if (si.getBinding().getUri().equals(SamlProtocol.POST_BINDING)) {
+        return si;
+      }
```

Comment:
We are but there was an iTest that was returning Redirect cause it didn't have post in its metadata. I added that check to make sure we get post

Reply:
Really? Shouldn't the IdP just be failing if this happens? It seems like this method (`getAssertionConsumerService`)would return `null` if the metadata doesn't have POST. From looking at all usages of this method in the code, it appears that the IdP ends up throwing some sort of exception when `null` is returned here.

Reply:
`getAssertionConsumerServiceInfoByUrl` does 
```
return new ServiceInfo(
          assertionConsumerServiceURL, 
          Binding.from(request.getProtocolBinding()), null);
```
at the end so that could return redirect. I wanted to continue trying other ways to find the correct url.